### PR TITLE
Fix: Clear properties of `NetworkRuleSet` after deleting `storage account network rules` resource

### PR DESCRIPTION
**Pull request by ms-zhenhua**
fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/15645
image

### CODE REVIEW

File: internal/services/storage/storage_account_network_rules_resource_test.go [hunk 1]
```diff
@@ -151,6 +151,34 @@ func TestAccStorageAccountNetworkRules_empty(t *testing.T) {
...
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.remove(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That("azurerm_storage_account.test").ExistsInAzure(r),
+			),
+		},
+		{
```
Comment (katbyte) [inline internal/services/storage/storage_account_network_rules_resource_test.go:172]: can we add an import step here?

Comment (ms-zhenhua) [reply internal/services/storage/storage_account_network_rules_resource_test.go:172]: Hi @katbyte , this step(**r.remove(data)**) will delete the **azurerm_storage_account_network_rules** resource. If we add **ImportStep()** here, the system will report error of **resource id not exist**.

Comment (katbyte) [reply internal/services/storage/storage_account_network_rules_resource_test.go:172]: Ah, in this case can we add a custom test step that checks to ensure the resource has actually been correctly deleted? thanks!

Comment (tombuildsstuff) [reply internal/services/storage/storage_account_network_rules_resource_test.go]: we'll want to confirm the Exists function does the same here too

Comment (ms-zhenhua) [reply internal/services/storage/storage_account_network_rules_resource_test.go]: Hi @katbyte , @tombuildsstuff , I have updated the test cases and all of them passed(Except **TestAccStorageAccountNetworkRules_SynapseAccess**, which is already failed in main branch). Kindly help review it again. Thanks.

**Review by katbyte — APPROVED** [review]
Thanks @ms-zhenhua - LGTM ❄️

**Pull request merged**